### DISCUSSION TITLE
swell.get.account() method returns null

### DISCUSSION THREAD

**Discussion by JoeMatkin** [swellstores/community#257] [Help] [answered]
I have logged in a user using swell.account.login(username, password) and successfully got a response with user data meaning I have successfully logged in.

But when I use swell.account.get() right after logging in or anyway in my app It returns null.

Bellow is my code:
```
import swell from 'lib/swellClient'
import { redirect } from 'next/navigation'

interface RequestBody {
  email: string
  password: string
}

export async function POST(request: Request) {
  const body: RequestBody = await request.json()
  console.log('login route body:', body)

  const user = await swell.account.login(body.email, body.password)
  const account = await swell.account.get()
  console.log('account:', account) // this receives null

  if (user) {
    const { password, ...userWithoutPass } = user
    const result = {
      ...userWithoutPass,
    }

    // TODO: Set login info to cookies and context api will use this to create userContext

    return new Response(JSON.stringify(user))
  } else return new Response(JSON.stringify(null))
}
```

**Comment by JoeMatkin** [comment]
Would really appreciate some help on this, thanks 

@ericingram

**Reply by JoeMatkin** [reply]
Still waiting for a response..

**Comment by sags95** [comment]
It's possible the session is not persisting on the server-side. Do you get the same issues when calling this outside of the route handler?

**Reply by JoeMatkin** [reply]
No, the problem doesn't exist outside of the api routes

**Comment by rodrigoddalmeida** [comment, marked as answer]
@JoeMatkin it looks like you are running swell.js in a server-side context - that means we can't rely properly on cookies to persist session data and make it available between requests. That could be the cause.

In your case, I would suggest a different approach for server-side, where you should call the `login` endpoint directly and store the session token in your app context. That way you can recover it and use in subsequent requests via Swell.js. So:

- You would POST `your_store.swell.store/api/account/login` with `{ email: ..., password: ... }` -  be sure to add the following headers to the request:

```js
{
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Basic ${swell.utils.base64Encode(YOUR_PUBLIC_KEY)}`,
};
```

- The response should include a `X-Session` header with a stateful token as a value. Store that. You can also pass it to a initialized Swell.js instance via `swell.options.session = SESSION_TOKEN` so requests made from the library are including the session token for authentication.

One thing you should take note: our session is stateful, meaning it can change value depending on the operation, params or endpoints that were requested - as a best practice and to avoid inconsistencies, you should always check the `X-Session` response header and store it to use in subsequent requests.

As an alternative, you can monkey-patch `cookies/getCookie` and `cookies/setCookie` in swell-js to intercept calls and store those values server-side - this is an interesting approach if you don't want to call our REST endpoints directly and just use Swell.js.

Let me know how it works.

**Reply by JoeMatkin** [reply]
Thanks for this, this was exactly the issue and all is working now.
